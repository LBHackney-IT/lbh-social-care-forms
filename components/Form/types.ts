/* eslint-disable @typescript-eslint/no-explicit-any */

import { InputHTMLAttributes } from 'react';
import {
  RegisterOptions,
  ValidateResult,
  UseFormMethods,
} from 'react-hook-form';
import TextInput from './TextInput/TextInput';

type EnhancedValidate = (
  data: any,
  formData: Record<string, any>
) => ValidateResult | Promise<ValidateResult>;

interface EnhancedRules extends Omit<RegisterOptions, 'validate'> {
  validate?: EnhancedValidate | Record<string, EnhancedValidate>;
}

type LabelSize = 's' | 'm' | 'l' | 'xl';
type Width = 30 | 20 | 10 | 5 | 4 | 3 | 2;
type GovGrid =
  | 'full'
  | 'half'
  | 'one-half'
  | 'one-third'
  | 'two-thirds'
  | 'one-quarter'
  | 'three-quarters';

export type ObjectOption = {
  text: string;
  value: string | number;
  disabled?: boolean;
};

export type Option = string | ObjectOption;

type FuncOption = (arg0: Record<string, string>) => Option[] | null;

export type Options = FuncOption | Option[];

export interface GenericField {
  name: string;
  label?: string;
  rules?: RegisterOptions;
  register?: UseFormMethods['register'];
  error?: {
    message?: string;
    type?: string;
  };
  required?: boolean | string;
  hint?: string;
  labelSize?: LabelSize;
  conditionalRender?: (arg0: Record<string, any>) => boolean;
  showConditionalGuides?: boolean;
  isMulti?: boolean;
  isMultiInit?: boolean;
  isMultiTrigger?: string;
  buttonStyle?: string;
}

export interface Checkbox extends GenericField {
  options?: Option[];
}

interface StepCheckbox extends Omit<Checkbox, 'options'> {
  component: 'Checkbox';
  options?: Options;
}

type TextInputType = GenericField & InputHTMLAttributes<HTMLInputElement>;

export interface TextInput extends TextInputType {
  inputClassName?: string;
  width?: Width;
}

interface StepTextInput extends TextInput {
  component: 'TextInput';
}

export type TextInputNoType = Omit<TextInput, 'type'>;

interface StepTextInputNoType extends TextInputNoType {
  component: 'NumberInput' | 'DatePicker' | 'PhoneInput' | 'EmailInput';
  defaultToday?: boolean;
}

type TextAreaType = GenericField & InputHTMLAttributes<HTMLTextAreaElement>;

export interface TextArea extends TextAreaType {
  width?: Width;
  rows?: number;
}

interface StepTextArea extends Omit<TextArea, 'rules'> {
  component: 'TextArea';
  rules?: EnhancedRules;
}

type SelectType = GenericField &
  Omit<InputHTMLAttributes<HTMLSelectElement>, 'onChange'>;

export interface Select extends SelectType {
  options: Option[];
  placeHolder?: string;
  ignoreValue?: boolean;
  isUnselectable?: boolean;
  width?: Width;
  govGrid?: GovGrid;
  onChange?: (arg0: string) => string | void;
}

interface StepSelect extends Omit<Select, 'options' | 'rules'> {
  component: 'Select';
  options: Options;
  rules?: EnhancedRules;
}

type RadiosType = GenericField & InputHTMLAttributes<HTMLInputElement>;

export interface Radios extends RadiosType {
  options?: Option[];
  isRadiosInline?: boolean;
  children?: React.ReactNode;
}

interface StepRadios extends Omit<Radios, 'options' | 'rules'> {
  component: 'Radios';
  options?: Options;
  rules?: EnhancedRules;
}

export interface ObjectInput extends GenericField {
  isMulti?: boolean;
  isInline?: boolean;
  summaryInline?: boolean;
  components: Array<formBasicComponentStep>;
}

interface StepObjectInput extends Omit<ObjectInput, 'rules'> {
  component: 'ObjectInput';
  rules?: EnhancedRules;
}

export interface AddressLookup extends GenericField {
  supportManualEntry?: boolean;
}

interface StepAddressLookup extends Omit<AddressLookup, 'rules'> {
  component: 'AddressLookup';
  rules?: EnhancedRules;
}

export interface Autocomplete extends Omit<GenericField, 'onChange'> {
  name: string;
  label: string;
  options: Option[];
  placeholder?: string;
  width?: Width;
  govGrid?: GovGrid;
  onChange?: (value: string | number | null) => void;
}

interface StepAutocomplete extends Omit<Autocomplete, 'options'> {
  component: 'Autocomplete';
  options: Options;
}

export interface DateInput extends GenericField {
  format?: 'ISO' | 'EU';
}

interface StepDateInput extends Omit<DateInput, 'rules'> {
  component: 'DateInput';
  rules?: EnhancedRules;
}

type formBasicComponentStep =
  | StepTextInput
  | StepTextInputNoType
  | StepTextArea
  | StepCheckbox
  | StepSelect
  | StepRadios
  | StepAddressLookup
  | StepAutocomplete
  | StepDateInput;

export type FormComponentStep =
  | formBasicComponentStep
  | StepObjectInput
  | JSX.Element;

export interface FormStep {
  id: string;
  title: string;
  components: Array<FormComponentStep>;
  isMulti?: boolean;
  conditionalRender?: (arg0: Record<string, unknown>) => boolean;
}

export interface FormConfig {
  title: string;
  path: string;
  successMessage?: string;
  steps: Array<FormStep>;
}
