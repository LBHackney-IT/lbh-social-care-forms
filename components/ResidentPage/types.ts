import { FlexibleAnswers } from 'data/flexibleForms/forms.types';

export enum WorkflowType {
  Assessment = 'Assessment',
  Review = 'Review',
  Reassessment = 'Reassessment',
  Historic = 'Historic',
}

export type Workflow = {
  id: string;
  type: WorkflowType;
  formId: string;
  answers: FlexibleAnswers;
  socialCareId: string;
  heldAt?: Date;
  teamAssignedTo?: string;
  assignedTo?: string;
  createdAt: Date;
  createdBy: string;
  submittedAt?: Date;
  submittedBy?: string;
  managerApprovedAt?: Date;
  managerApprovedBy?: string;
  needsPanelApproval: boolean;
  panelApprovedAt?: Date;
  panelApprovedBy?: string;
  workflowId?: string;
  reviewBefore?: Date;
  linkToOriginal?: string;
  acknowledgedAt?: Date;
  acknowledgedBy?: string;
  acknowledgingTeam?: string;
  discardedAt?: Date;
  discardedBy?: string;
  updatedAt: Date;
  updatedBy?: string;
  episodeId?: string;
  form?: {
    id: string;
    name: string;
  };
};
