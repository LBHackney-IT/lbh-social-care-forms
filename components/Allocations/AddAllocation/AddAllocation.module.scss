.workerLink {
  margin-top: 1rem;
  svg {
    margin-right: 10px;
  }

  &:hover svg rect {
    fill: lbh-colour('lbh-link-hover');
  }
  &:active svg rect {
    fill: lbh-colour('lbh-link-active');
  }
  &:focus svg rect {
    fill: lbh-colour('lbh-text');
  }
}
