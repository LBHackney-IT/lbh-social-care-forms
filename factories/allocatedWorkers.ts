import { Factory } from 'fishery';

import { Allocation } from 'types';

export const allocationFactory = Factory.define<Allocation>(({ sequence }) => ({
  id: sequence,
  allocatedWorkerTeam: 'Safeguarding and Reviewing Team',
  allocatedWorker: 'Officer Name',
  allocationEndDate: '2019-03-28 00:00:00',
  allocationStartDate: '2019-03-28 00:00:00',
  workerType: 'Consultant Social Worker',
  personId: 1234,
  caseStatus: 'Open',
  personName: 'foo',
  personAddress: 'the address',
  personDateOfBirth: '2020-03-20',
  ragRating: 'amber',
}));

export const mockedAllocation = allocationFactory.build();
export const mockedAllocations = [allocationFactory.build()];
