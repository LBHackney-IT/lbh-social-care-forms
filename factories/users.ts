import { Factory } from 'fishery';

import { User } from 'types';

export const userFactory = Factory.define<User>(() => ({
  name: 'foo',
  hasAdminPermissions: true,
  hasChildrenPermissions: true,
  hasAdultPermissions: true,
  hasDevPermissions: false,
  isAuditable: false,
  email: 'foo@bar.com',
  permissionFlag: 'A',
  isAuthorised: true,
  isInWorkflowsPilot: false,
  isInSafeguardingReviewing: false,
  isInPlacementManagementUnit: false,
}));

export const mockedOnlyAdultUser = userFactory.build({
  hasAdminPermissions: false,
  hasChildrenPermissions: false,
  hasDevPermissions: false,
});

export const mockedOnlyChildUser = userFactory.build({
  hasAdminPermissions: false,
  hasAdultPermissions: false,
  hasDevPermissions: false,
});

export const mockedAdminUser = userFactory.build({
  hasDevPermissions: false,
});

export const mockedUser = userFactory.build();

export const mockedUserInWorkflowsPilot = userFactory.build({
  isInWorkflowsPilot: true,
});
