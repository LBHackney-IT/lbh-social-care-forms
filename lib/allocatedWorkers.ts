import axios from 'axios';
import * as yup from 'yup';
import isPast from 'date-fns/isPast';
import parseISO from 'date-fns/parseISO';

import type { Allocation, AllocationData } from 'types';

const ENDPOINT_API = process.env.ENDPOINT_API;
const AWS_KEY = process.env.AWS_KEY;

interface AllocationsParams {
  mosaic_id?: number;
  worker_id?: number;
  [key: string]: unknown;
}

const filterClosedAllocations = (allocations: Allocation[]) =>
  allocations?.filter(
    ({ caseStatus, allocationEndDate }) =>
      caseStatus?.toLowerCase() !== 'closed' &&
      (!allocationEndDate || !isPast(parseISO(allocationEndDate)))
  );

export const getAllocations = async (
  params: AllocationsParams,
  showOnlyOpen = true
): Promise<AllocationData> => {
  const { data } = await axios.get(`${ENDPOINT_API}/allocations`, {
    headers: { 'x-api-key': AWS_KEY },
    params,
  });
  return showOnlyOpen
    ? {
        ...data,
        allocations: filterClosedAllocations(data.allocations),
      }
    : data;
};

export const getResidentAllocatedWorkers = (
  mosaic_id: number,
  params?: AllocationsParams
): Promise<AllocationData> => getAllocations({ mosaic_id, ...params });

export const getResidentAllocation = async (
  mosaic_id: number,
  allocation_id: number,
  params?: AllocationsParams
): Promise<Allocation | undefined> => {
  const showOnlyOpen = false;
  const data = await getAllocations({ mosaic_id, ...params }, showOnlyOpen);
  return data?.allocations?.find(({ id }) => id === allocation_id);
};
export const getAllocationsByWorker = async (
  worker_id: number,
  params?: AllocationsParams
): Promise<AllocationData> => getAllocations({ worker_id, ...params });

const deleteAllocationSchema = yup.object({
  id: yup.number().required().positive().integer(),
  deallocationReason: yup.string().required(),
  deallocationDate: yup.string().required(),
  createdBy: yup.string().email().required(),
});

export const deleteAllocation = async (
  params: yup.InferType<typeof deleteAllocationSchema>
): Promise<Record<string, unknown>> => {
  const body = await deleteAllocationSchema.validate(params);
  const { data } = await axios.patch(`${ENDPOINT_API}/allocations`, body, {
    headers: { 'Content-Type': 'application/json', 'x-api-key': AWS_KEY },
  });
  return data;
};

const addAllocatedWorkerSchema = yup.object({
  mosaicId: yup.number().required().positive().integer(),
  allocatedTeamId: yup.number().required().integer(),
  allocatedWorkerId: yup.number().integer().nullable(),
  allocationStartDate: yup.string().required(),
  ragRating: yup.string().required(),
  createdBy: yup.string().email().required(),
});

export const addAllocatedWorker = async (
  mosaicId: yup.InferType<typeof addAllocatedWorkerSchema>['mosaicId'],
  params: Omit<yup.InferType<typeof addAllocatedWorkerSchema>, 'mosaicId'>
): Promise<Record<string, unknown>> => {
  const body = await addAllocatedWorkerSchema.validate({ mosaicId, ...params });

  const { data } = await axios.post(`${ENDPOINT_API}/allocations`, body, {
    headers: { 'Content-Type': 'application/json', 'x-api-key': AWS_KEY },
  });
  return data;
};

const addWorkerAllocationSchema = yup.object({
  allocationId: yup.number().integer().nullable(),
  allocatedTeamId: yup.number().integer().nullable(),
  allocatedWorkerId: yup.number().integer().nullable(),
  allocationStartDate: yup.string().required(),
});

export const addWorkerAllocation = async (
  mosaicId: number,
  params: Omit<yup.InferType<typeof addWorkerAllocationSchema>, 'mosaicId'>
): Promise<Record<string, unknown>> => {
  const body = await addWorkerAllocationSchema.validate({
    mosaicId,
    ...params,
  });
  const { data } = await axios.post(`${ENDPOINT_API}/allocations`, body, {
    headers: { 'Content-Type': 'application/json', 'x-api-key': AWS_KEY },
  });

  return data;
};
