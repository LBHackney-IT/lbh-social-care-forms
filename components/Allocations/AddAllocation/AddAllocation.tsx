import React, { useState, useCallback, useEffect } from 'react';
import { useRouter } from 'next/router';
import { useForm } from 'react-hook-form';
import { Autocomplete } from 'components/Form';
import Button from 'components/Button/Button';
import Spinner from 'components/Spinner/Spinner';
import ErrorMessage from 'components/ErrorMessage/ErrorMessage';
import { useTeams, useTeamWorkers } from 'utils/api/allocatedWorkers';
import { AgeContext } from 'types';
import DateInput from 'components/Form/DateInput/DateInput';
import Radios from 'components/Form/Radios/Radios';
import SelectWorker from './SelectWorker';
import { allocateResident } from '../../../lib/allocations';

interface Props {
  personId: number;
  ageContext: AgeContext;
}

const AddAllocation = ({ personId, ageContext }: Props): React.ReactElement => {
  const [postError, setPostError] = useState<boolean | null>();
  const [postLoading, setPostLoading] = useState<boolean>();
  const [workerAllocation, setWorkerAllocation] = useState<boolean>(false);
  const { query, push, replace, pathname } = useRouter();
  const { handleSubmit, control, errors } = useForm({
    defaultValues: query,
  });
  const { teamId } = query as { teamId?: number };
  const { data: { teams } = {}, error: errorTeams } = useTeams({ ageContext });
  const { data: workers, error: errorWorkers } = useTeamWorkers(teamId);
  const addWorker = useCallback(
    async ({ workerId, allocationStartDate }) => {
      setPostLoading(true);
      setPostError(null);
      try {
        await allocateResident({
          personId: Number(personId),
          allocatedTeamId: Number(teamId),
          workerId: Number(workerId),
          createdBy: Number(),
          summary: '',
          carePackage: '',
          ragRating: 'purple',
          allocationDate: allocationStartDate,
        });
        push(`/people/${personId}`);
      } catch (e) {
        setPostError(true);
      }
      setPostLoading(false);
    },
    [personId, teamId, push]
  );
  useEffect(() => {
    setPostError(null);
  }, [query]);
  if (errorTeams || errorWorkers || postError) {
    return <ErrorMessage />;
  }
  if (!teams) {
    return <Spinner />;
  }
  return (
    <form role="form" onSubmit={handleSubmit(addWorker)}>
      {teams && (
        <Autocomplete
          name="teamId"
          label="Select a team"
          labelSize="m"
          placeholder="Select or type team name"
          control={control}
          options={teams.map(({ id, name }) => ({
            value: id,
            text: name,
          }))}
          onChange={(value: string | number | null) => {
            replace(
              {
                pathname,
                query: { id: personId, teamId: value },
              },
              undefined,
              { scroll: false }
            );
          }}
          required
        />
      )}

      <Radios
        name="priority"
        label="Choose a priority rating"
        options={[
          'Urgent priority',
          'High priority',
          'Medium priority',
          'Low priority',
        ]}
        onChange={(elm) => {
          console.log('change', elm.target.value);
        }}
        required
      />

      <DateInput
        label="Select an allocation date"
        labelSize="s"
        name="allocationStartDate"
        error={errors.allocationStartDate}
        control={control}
        required
      />

      {!workerAllocation ? (
        <a className="lbh-link" onClick={() => setWorkerAllocation(true)}>
          <svg
            width="15"
            height="15"
            viewBox="0 0 15 15"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect x="7" width="1" height="15" fill="#025EA6" />
            <rect
              x="15"
              y="7"
              width="1"
              height="15"
              transform="rotate(90 15 7)"
              fill="#025EA6"
            />
          </svg>
          View
        </a>
      ) : (
        <></>
      )}

      {workerAllocation && workers?.workers ? (
        <>
          <SelectWorker
            records={workers?.workers}
            callback={(value: any) => console.log(value)}
          />
          <a className="lbh-link" onClick={() => setWorkerAllocation(false)}>
            <svg
              width="15"
              height="15"
              viewBox="0 0 15 15"
              fill="none"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect x="7" width="1" height="15" fill="#025EA6" />
              <rect
                x="15"
                y="7"
                width="1"
                height="15"
                transform="rotate(90 15 7)"
                fill="#025EA6"
              />
            </svg>
            hide
          </a>
        </>
      ) : (
        <></>
      )}

      <Button label="Continue" type="submit" disabled={postLoading} />
      {teamId && (!teams || !workers) && <Spinner />}
    </form>
  );
};

export default AddAllocation;
