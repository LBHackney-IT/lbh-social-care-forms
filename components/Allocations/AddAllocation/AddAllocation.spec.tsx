import { act, fireEvent, render } from '@testing-library/react';
import { UserContext } from 'components/UserContext/UserContext';
import { mockedUser } from 'factories/users';
import AddAllocation from './AddAllocation';
import * as allocatedWorkerAPI from 'utils/api/allocatedWorkers';
import { AgeContext } from 'types';
import { format } from 'date-fns';

jest.mock('next/router', () => ({
  useRouter: () => ({
    query: { teamId: '3' },
    asPath: 'path',
    push: jest.fn(),
    replace: jest.fn(),
  }),
}));
jest.mock('components/Spinner/Spinner', () => () => 'MockedSpinner');

describe(`AddAllocation`, () => {
  jest.spyOn(allocatedWorkerAPI, 'useTeams').mockImplementation(() => ({
    data: {
      teams: [
        { id: 1, name: 'Team 1' },
        { id: 2, name: 'Team 2' },
        { id: 3, name: 'Team 3' },
      ],
    },
    revalidate: jest.fn(),
    mutate: jest.fn(),
    isValidating: false,
  }));
  jest.spyOn(allocatedWorkerAPI, 'useTeamWorkers').mockImplementation(() => ({
    workers: [
      {
        id: 9,
        firstName: 'Worker',
        lastName: 'A',
        allocationCount: 3,
        role: 'role_a',
        email: 'a@email.com',
        teams: [],
      },
      {
        id: 8,
        firstName: 'Worker',
        lastName: 'B',
        allocationCount: 0,
        role: 'role_b',
        email: 'b@email.com',
        teams: [],
      },
      {
        id: 7,
        firstName: 'Worker',
        lastName: 'C',
        allocationCount: 1,
        role: 'role_c',
        email: 'c@email.com',
        teams: [],
      },
    ],
    revalidate: jest.fn(),
    mutate: jest.fn(),
    isValidating: false,
  }));
  const props = {
    personId: 123,
    ageContext: 'A' as AgeContext,
    control: { defaultValuesRef: { current: { name: 'teamId' } } },
  };

  it('should not show "Allocate a worker" if there is no team selected', async () => {
    const { queryByText } = render(
      <UserContext.Provider
        value={{
          user: mockedUser,
        }}
      >
        <AddAllocation {...props} />
      </UserContext.Provider>
    );
    const allocate_worker_link = queryByText('+ Allocate a worker');
    expect(allocate_worker_link).not.toBeInTheDocument();
  });
  xit('should show "Allocate a worker" if a team selected', async () => {
    jest.spyOn(allocatedWorkerAPI, 'useTeamWorkers').mockImplementation(() => ({
      workers: [
        {
          id: 9,
          firstName: 'Worker',
          lastName: 'A',
          allocationCount: 3,
          role: 'role_a',
          email: 'a@email.com',
          teams: [],
        },
      ],
      revalidate: jest.fn(),
      mutate: jest.fn(),
      isValidating: false,
    }));

    const { getByTestId } = render(
      <UserContext.Provider
        value={{
          user: mockedUser,
        }}
      >
        <AddAllocation {...props} />
      </UserContext.Provider>
    );
    const teamAutocomplete = getByTestId('teamId');

    await act(async () => {
      fireEvent.click(teamAutocomplete);
      fireEvent.change(teamAutocomplete, { target: { value: 'Team 3' } });
    });
    await act(async () => {
      fireEvent.click(getByTestId('teamId'));
      expect(teamAutocomplete).toHaveValue('Team 3');
    });

    const allocate_worker_link = getByTestId('allocate_worker_link');
    expect(allocate_worker_link).toBeInTheDocument();
  });

  it('should render and submit correctly', async () => {
    jest.spyOn(allocatedWorkerAPI, 'addAllocatedWorker');

    const { getByText, getByRole, getByTestId } = render(
      <UserContext.Provider
        value={{
          user: mockedUser,
        }}
      >
        <AddAllocation {...props} />
      </UserContext.Provider>
    );
    const teamAutocomplete = getByTestId('teamId');
    await act(async () => {
      fireEvent.click(teamAutocomplete);
    });
    await act(async () => {
      fireEvent.click(getByText('Team 3'));
    });
    await act(async () => {
      fireEvent.click(getByText('Medium priority'));
    });
    await act(async () => {
      fireEvent.submit(getByRole('form'));
    });

    expect(allocatedWorkerAPI.addAllocatedWorker).toHaveBeenCalled();
    expect(allocatedWorkerAPI.addAllocatedWorker).toHaveBeenCalledWith(123, {
      allocatedTeamId: 3,
      allocationStartDate: format(new Date(), 'yyyy-MM-dd'),
      ragRating: 'amber',
    });
  });
});
