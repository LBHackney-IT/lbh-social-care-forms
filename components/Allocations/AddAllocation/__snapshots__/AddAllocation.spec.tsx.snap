// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddAllocation should render properly 1`] = `
<DocumentFragment>
  <form
    role="form"
  >
    <div
      aria-expanded="true"
      aria-haspopup="listbox"
      aria-labelledby="downshift-0-label"
      aria-owns="downshift-0-menu"
      class="lbh-form-group govuk-form-group"
      role="combobox"
    >
      <label
        class="govuk-label lbh-label"
        id="downshift-0-label"
      >
        Select a team 
        <span
          class="lbh-required"
        >
          *
        </span>
      </label>
      <div
        class="inputContainer"
      >
        <input
          aria-autocomplete="list"
          aria-controls="downshift-0-menu"
          aria-labelledby="downshift-0-label"
          autocomplete="off"
          class="lbh-input govuk-input govuk-input--width-20 "
          data-testid="teamId"
          id="teamId"
          placeholder="Select or type team name"
          value="Team 3"
        />
        <button
          aria-haspopup="true"
          aria-label="toggle menu"
          class="button"
          data-toggle="true"
          role="button"
          type="button"
        >
          <svg
            fill="none"
            height="10"
            viewBox="0 0 17 10"
            width="17"
          >
            <path
              d="M2 1.5L8.5 7.5L15 1.5"
              stroke="#0B0C0C"
              stroke-width="3"
            />
          </svg>
        </button>
        <ul
          aria-labelledby="downshift-0-label"
          class="list"
          id="downshift-0-menu"
          role="listbox"
          style="border-bottom: 1px solid #aaa;"
        >
          <li
            aria-selected="false"
            class="option"
            data-testid="teamId_0"
            id="downshift-0-item-0"
            role="option"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 255, 255);"
          >
            Team 3
          </li>
        </ul>
      </div>
    </div>
    <div
      class="govuk-form-group lbh-form-group"
    >
      <label
        class="lbh-label govuk-label"
        for="priority"
      >
        Choose a priority rating 
        <span
          class="govuk-required"
        >
          *
        </span>
      </label>
      <div
        class="govuk-radios lbh-radios"
      >
        <div
          class="govuk-radios__item"
        >
          <input
            class="govuk-radios__input"
            id="priority_purple"
            name="priority"
            type="radio"
            value="purple"
          />
          <label
            class="govuk-label govuk-radios__label"
            for="priority_purple"
          >
            Urgent priority
          </label>
        </div>
        <div
          class="govuk-radios__item"
        >
          <input
            class="govuk-radios__input"
            id="priority_red"
            name="priority"
            type="radio"
            value="red"
          />
          <label
            class="govuk-label govuk-radios__label"
            for="priority_red"
          >
            High priority
          </label>
        </div>
        <div
          class="govuk-radios__item"
        >
          <input
            class="govuk-radios__input"
            id="priority_amber"
            name="priority"
            type="radio"
            value="amber"
          />
          <label
            class="govuk-label govuk-radios__label"
            for="priority_amber"
          >
            Medium priority
          </label>
        </div>
        <div
          class="govuk-radios__item"
        >
          <input
            class="govuk-radios__input"
            id="priority_green"
            name="priority"
            type="radio"
            value="green"
          />
          <label
            class="govuk-label govuk-radios__label"
            for="priority_green"
          >
            Low priority
          </label>
        </div>
        <div
          class="govuk-radios__item"
        >
          <input
            class="govuk-radios__input"
            id="priority_white"
            name="priority"
            type="radio"
            value="white"
          />
          <label
            class="govuk-label govuk-radios__label"
            for="priority_white"
          >
            No priority
          </label>
        </div>
      </div>
    </div>
    <div
      class="govuk-form-group lbh-form-group"
    >
      <label
        class="govuk-label lbh-label"
        for="allocationStartDate"
      >
        Select an allocation date 
        <span
          class="govuk-required"
        >
          *
        </span>
      </label>
      <input
        class="govuk-input lbh-input govuk-input--width-10"
        data-testid="allocationStartDate"
        id="allocationStartDate"
        name="allocationStartDate"
        type="date"
        value="2022-03-18"
      />
    </div>
    <br />
    <button
      class="lbh-button govuk-button"
      data-module="govuk-button"
      disabled=""
      type="submit"
    >
      Continue
    </button>
    MockedSpinner
  </form>
</DocumentFragment>
`;
